### PR TITLE
Fix filter commit label

### PR DESCRIPTION
**Pull request by pmiossec**
Changes proposed in this pull request:
- Fix filter commit label which was displayed  like "Commit..." instead of "Commit" because the transation was the one of the "commands" menu due to the same control name (a real problem in french translation for example).
- Use of a much detailed label to better tell what the filter do (I, myself, never know that the filter was also used to search in sha1)
 
Screenshots before and after (if PR changes UI):
- before:

image

- after:

image

### CODE REVIEW

File: GitUI/FilterRevisionsHelper.cs [hunk 1]
```diff
@@ -29,10 +29,10 @@ public FilterRevisionsHelper()
...
             // commitToolStripMenuItem1
             //
-            _commitToolStripMenuItem.Checked = true;
-            _commitToolStripMenuItem.CheckOnClick = true;
-            _commitToolStripMenuItem.Name = "commitToolStripMenuItem1";
-            _commitToolStripMenuItem.Text = "Commit";
+            _commitFilterToolStripMenuItem.Checked = true;
+            _commitFilterToolStripMenuItem.CheckOnClick = true;
+            _commitFilterToolStripMenuItem.Name = "commitToolStripMenuItem1";
+            _commitFilterToolStripMenuItem.Text = "Commit message and Sha1";
```
Comment (gerhardol) [inline GitUI/FilterRevisionsHelper.cs]: SHA-1 (and SHA) is used in similar commands

Comment (sharwell) [reply GitUI/FilterRevisionsHelper.cs]: :bulb: I would actually use 'hash' instead of 'SHA'

Comment (RussKie) [reply GitUI/FilterRevisionsHelper.cs]: I've heard some people use "commish" for "commit hash".
Either way we need to be consistent, and if we currently use SHA and SHA-1 then we need to continue with this.
We can reword it to "hash" or "commit hash" (or even "commish") separately before v3 is released

Comment (pmiossec) [reply GitUI/FilterRevisionsHelper.cs]: They surely use "commitish" ;-) but that is larger than just "commit hash" and refer to everything that end up pointing to a commit.

So, I think it is not understandable by most and also not exactly what we filter here (except if I'm wrong).

I like "hash" (or "SHA") which are more future proof :-)

I will update to "SHA" if that's we use at the moment.

Comment (pmiossec) [reply GitUI/FilterRevisionsHelper.cs]: finally, I have choosen "hash" because that's what is used in the commit details tab and in the "copy to clipboard" menu.
